src/Compilers/CSharp/Portable/Symbols/Synthesized/InjectedNonNullTypesAttributeSymbol.cs

```diff
@@ -0,0 +1,133 @@
+    {
+        private ImmutableArray<CSharpAttributeData> _lazyCustomAttributes;
+
+        private InjectedNonNullTypesAttributeSymbol(
+            AttributeDescription description,
+            NamespaceSymbol containingNamespace,
+            CSharpCompilation compilation,
+            Func<CSharpCompilation, NamedTypeSymbol, DiagnosticBag, ImmutableArray<MethodSymbol>> getConstructors)
+            : base(description, containingNamespace, compilation, getConstructors)
+        {
+        }
+
+        public static InjectedNonNullTypesAttributeSymbol Create(NamespaceSymbol containingNamespace)
+        {
+            return new InjectedNonNullTypesAttributeSymbol(AttributeDescription.NonNullTypesAttribute, containingNamespace, containingNamespace.DeclaringCompilation, makeConstructor);
+
+            ImmutableArray<MethodSymbol> makeConstructor(CSharpCompilation compilation, NamedTypeSymbol containingType, DiagnosticBag diagnostics)
+            {
+                var boolType = compilation.GetSpecialType(SpecialType.System_Boolean);
+
+                Binder.ReportUseSiteDiagnostics(boolType, diagnostics, Location.None);
+
+                var boolWithAnnotations = TypeSymbolWithAnnotations.Create(boolType);
+                // PROTOTYPE constructor should save the parameter into a field (for users of reflection)
+                return ImmutableArray.Create<MethodSymbol>(
+                    new NonNullTypesAttributeConstructorSymbol(
+                        containingType,
+                        m => ImmutableArray.Create(SynthesizedParameterSymbol.Create(m, boolWithAnnotations, 0, ConstantValue.True, name: "flag"))));
+            }
+        }
+
+        internal override void AddDiagnostics(DiagnosticBag recipient)
+        {
+            // pull on the attributes to collect their diagnostics too
+            _ = GetAttributes();
+            recipient.AddRange(_diagnostics);
+        }
+
+        public override ImmutableArray<CSharpAttributeData> GetAttributes()
+        {
+            if (_lazyCustomAttributes.IsDefault)
+            {
+                ImmutableInterlocked.InterlockedInitialize(ref _lazyCustomAttributes, MakeAttributes());
+            }
+            return _lazyCustomAttributes;
+        }
+
+        /// <summary>
+        /// Adds an `[AttributeUsage(AttributeTargets.Class | ...)]` (if possible) and captures any diagnostics in the process.
+        /// </summary>
+        private ImmutableArray<CSharpAttributeData> MakeAttributes()
+        {
+            var ctor = (MethodSymbol)Binder.GetWellKnownTypeMember(DeclaringCompilation, WellKnownMember.System_AttributeUsageAttribute__ctor, _diagnostics, Location.None);
+            if (ctor is null)
+            {
+                // member is missing
+                return ImmutableArray<CSharpAttributeData>.Empty;
+            }
+
+            Binder.ReportUseSiteDiagnostics(ctor, _diagnostics, Location.None);
```

Comment:
>Binder.ReportUseSiteDiagnostics(ctor, _diagnostics, Location.None); [](start = 12, length = 67)

Doesn't ```Binder.GetWellKnownTypeMember``` take care of this?

Reply:
I didn't manage to hit this with a test, but I fixed this anyways. 
The few ways I know to generate use-site diagnostics didn't work:
- Obsolete is not reported for well-known types
- I couldn't create a missing type because we're dealing with corlib

---
In reply to: [216092257](https://github.com/dotnet/roslyn/pull/29180#discussion_r216092257) [](ancestors = 216092257)